libraries/chain/asset_evaluator.cpp

```diff
@@ -308,6 +329,13 @@ void_result asset_update_evaluator::do_evaluate(const asset_update_operation& o)
    for( auto id : o.new_options.blacklist_authorities )
       d.get_object(id);
 
+   if(now <= HARDFORK_BSIP_81_TIME) {
+      // Taker fees should be zero until activation of BSIP81
+      FC_ASSERT(!o.new_options.extensions.value.taker_fee_percent.valid()
+                || *o.new_options.extensions.value.taker_fee_percent == 0,
+                "Simple maker-taker fees are not yet activated");
+   }
```

Comment:
I think we can call `check_asset_options_hf_bsip81()` function directly to avoid duplicate code.